src/test/java/io/visual_regression_tracker/sdk_java/VisualRegressionTrackerTest.java

```diff
@@ -49,209 +66,183 @@ public void setup() {
+        assertThat(request.getHeader(VisualRegressionTracker.apiKeyHeaderName), is(config.getApiKey()));
+        assertThat(request.getBody().readUtf8(), is(gson.toJson(testRunRequest)));
+        assertThat(gson.toJson(result), is(gson.toJson(testRunResponse)));
     }
 
-    @Test
+    @Test(expectedExceptions = TestRunException.class,
+            expectedExceptionsMessageRegExp = "Visual Regression Tracker has not been started")
     public void submitTestRunShouldThrowIfNotStarted() throws IOException {
-        VisualRegressionTracker vrtMocked = Mockito.mock(VisualRegressionTracker.class);
-        Mockito.when(vrtMocked.isStarted()).thenReturn(false);
+        when(vrtMocked.isStarted()).thenReturn(false);
 
-        Mockito.doCallRealMethod().when(vrtMocked).submitTestRun(Mockito.anyString(), Mockito.any(), Mockito.any());
-        String exceptionMessage = "";
-        try {
-            vrtMocked.submitTestRun("name", null, null);
-        } catch (TestRunException ex) {
-            exceptionMessage = ex.getMessage();
-        }
-        MatcherAssert.assertThat(exceptionMessage, CoreMatchers.is("Visual Regression Tracker has not been started"));
+        doCallRealMethod().when(vrtMocked).submitTestRun(anyString(), any(), any());
+        vrtMocked.submitTestRun("name", null, null);
     }
 
     @DataProvider(name = "trackErrorCases")
     public Object[][] trackErrorCases() {
-        return new Object[][]{
+        return new Object[][] {
                 {
                         TestRunResponse.builder()
-                                .url("https://someurl.com/test/123123")
-                                .status(TestRunStatus.UNRESOLVED)
+                                       .url("https://someurl.com/test/123123")
+                                       .status(TestRunStatus.UNRESOLVED)
                                 .build(),
                         "Difference found: https://someurl.com/test/123123"
                 },
                 {
                         TestRunResponse.builder()
-                                .url("https://someurl.com/test/123123")
-                                .status(TestRunStatus.NEW)
+                                       .url("https://someurl.com/test/123123")
+                                       .status(TestRunStatus.NEW)
                                 .build(),
                         "No baseline: https://someurl.com/test/123123"
                 }
         };
     }
 
-    @Test(dataProvider = "trackErrorCases")
+    @Test(dataProvider = "trackErrorCases",
+            expectedExceptions = TestRunException.class,
+            expectedExceptionsMessageRegExp = "^(Difference found: https://someurl.com/test/123123|No baseline: https://someurl.com/test/123123)$")
     public void trackShouldThrowException(TestRunResponse testRunResponse, String expectedExceptionMessage) throws IOException {
```

Comment:
could we make sure that each case is throwing expected message instead of any of them?

Reply:
In particular implementation - no. I can split it into two cases.

Reply:
np, better like this
another test with logs is checking message so should be fine